x-pack/plugins/security_solution/server/lib/detection_engine/routes/fleet/get_installed_integrations/installed_integration_set.ts

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { flatten } from 'lodash';
+import { PackagePolicy, RegistryPackage } from '@kbn/fleet-plugin/common';
+import {
+  InstalledIntegration,
+  InstalledIntegrationArray,
+  InstalledIntegrationBasicInfo,
+  InstalledPackage,
+  InstalledPackageArray,
+  InstalledPackageBasicInfo,
+} from '../../../../../../common/detection_engine/schemas/common';
+
+export interface IInstalledIntegrationSet {
+  addPackagePolicy(policy: PackagePolicy): void;
+  addRegistryPackage(registryPackage: RegistryPackage): void;
+
+  getPackages(): InstalledPackageArray;
+  getIntegrations(): InstalledIntegrationArray;
+}
+
+type PackageMap = Map<string, PackageInfo>;
+
+interface PackageInfo extends InstalledPackageBasicInfo {
+  integrations: Map<string, InstalledIntegrationBasicInfo>;
+}
+
+export const createInstalledIntegrationSet = (): IInstalledIntegrationSet => {
```

Comment:
As with the route, may be good to have a brief summary following the logic/construction of this data.